app/src/main/java/org/glucosio/android/tools/ReadingToCSV.java

```diff
@@ -30,22 +30,15 @@
 import java.io.File;
 import java.io.FileOutputStream;
 import java.io.OutputStreamWriter;
-import java.util.ArrayList;
+import java.util.List;
 
 import io.realm.Realm;
 
 public class ReadingToCSV {
 
-    private Context context;
-
-    public ReadingToCSV(Context mContext) {
-        this.context = mContext;
-    }
-
-
-    public String createCSVFile(Realm realm, final ArrayList<GlucoseReading> readings, String um) {
+    public static String createCSVFile(Context context, Realm realm, final List<GlucoseReading> readings, String um) {
```

Comment:
I always worry about performance, this can of course be tested if we change the `AsyncTask` to a Factory class/public method (as it should be). 
With the current setup, no you can't because you are receiving the clicked event and create the `AsyncTask` there, not a good pattern. And the fix _per se_ would get verbose, this would be my first contribution I don't think is a good atitude and change a lot, it may confuse/upset current contributors.

> Every kitten grows up to be a cat. They seem so harmless at first--small, quiet, lapping up their saucer of milk. But once their claws get long enough, they draw blood, sometimes from the hand that feeds them.

That being said, static elements are awesome when used right, they go both ways, if we ignore performance we'll have the same problems ;).